src/zgossip.c

```diff
@@ -274,7 +292,12 @@ server_method (server_t *self, const char *method, zmsg_t *msg)
     if (streq (method, "CONNECT")) {
         char *endpoint = zmsg_popstr (msg);
         assert (endpoint);
+#ifdef CZMQ_BUILD_DRAFT_API
+        const char *public_key = zmsg_popstr (msg);
+        server_connect (self, endpoint, public_key);
+#else
         server_connect (self, endpoint);
```

Comment:
keeping server_connect the same, here just pass NULL in this branch of the ifdef, so the code should be simpler

Reply:
ack, ty for this- my C is not so .. sharp :) appreciate it.

Reply:
err you mean straight up this:

```
if (streq (method, "CONNECT")) {
        char *endpoint = zmsg_popstr (msg);
        assert (endpoint);
        const char *public_key = zmsg_popstr (msg);
        server_connect (self, endpoint, public_key);
        zstr_free (&endpoint);
    }
```

(i didn't wanna colossally mess up someone's system while we were hammering this idea out, so everything defaulted to verbose-seperate-out, figured this would re-factor out as it was tested and  we became comfortable with it..)

Reply:
Yeah that should work even better - there shouldn't be a third string in the CONNECT message so popstr will return NULL